java/kotlin-extractor/dev/wrapper.py

```diff
@@ -122,10 +122,13 @@ def forward(tool, forwarded_opts):
     if platform.system() == "Windows":
         tool = tool.with_suffix(".bat")
     assert tool.exists(), f"{tool} not found"
-    args = [tool]
-    args.extend(forwarded_opts)
-    ret = subprocess.run(args).returncode
-    sys.exit(ret)
+    cmd = [tool] + forwarded_opts
+    if platform.system() == "Windows":
+        # kotlin bat script is pretty sensible to unquoted args on windows
+        ret = subprocess.run(" ".join(f'"{a}"' for a in cmd)).returncode
+        sys.exit(ret)
+    else:
+        os.execv(cmd[0], cmd)
```

Comment:
This seems like a substantial change for the non-windows case, despite the windows-only implication of the PR title. Is that intentional?

Reply:
(if you say the `subprocess.run` and `execv` approaches  are equivalent in this case, then 👍🏻 )

Reply:
you are right: this is an optimization on POSIX, where we start one less process. The only reason to not use `execv` before was because it has a very bad behaviour on windows (it starts the provided process asynchronously). Once I got  to take a different code path on the different platforms in any case, I thought to sneak in the optimization too. Is it ok if I just change the PR title?

Reply:
> (if you say the subprocess.run and execv approaches are equivalent in this case, then 👍🏻 )

Yes, they are equivalent on POSIX, minus the fact that we don't spawn a separate process but we replace the current one (which is something that is just not possible on Windows)